### ISSUE TITLE
Format of ETag is bad when quoted or unquoted

### ISSUE THREAD

**Issue by lukepuplett** [Azure/azure-storage-net#904]
### Which service(blob, file, queue, table) does this issue concern?
Table

### Which version of the SDK was used?
9.3.3

### Which platform are you using? (ex: .NET Core 2.1)
.NET Core 2.2

### What problem was encountered?
When performing a merge operation:

When `entity.ETag = "\"0\""` the following is encountered.

```
Microsoft.WindowsAzure.Storage.StorageException: Bad Request
      at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteAsyncInternal[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext, CancellationToken token) in C:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\WindowsRuntime\Core\Executor\Executor.cs:line 315
```

And when `entity.ETag = "0"` the following is encountered.

```
FormatException: The format of value '0' is invalid.
      System.Net.Http.Headers.HttpHeaderParser.ParseValue(string value, object storeValue, ref int index)
      System.Net.Http.Headers.HttpHeaders.ParseAndAddValue(HeaderDescriptor descriptor, HeaderStoreItemInfo info, string value)
      System.Net.Http.Headers.HttpHeaders.Add(HeaderDescriptor descriptor, string value)
      Microsoft.WindowsAzure.Storage.Table.Protocol.TableOperationHttpRequestMessageFactory.BuildRequestForTableOperation<T>(RESTCommand<T> cmd, Uri uri, UriQueryBuilder builder, Nullable<int> timeout, TableOperation operation, CloudTableClient client, HttpContent content, OperationContext ctx, TablePayloadFormat payloadFormat, ICanonicalizer canonicalizer, StorageCredentials credentials) in TableOperationHttpRequestFactory.cs
      Microsoft.WindowsAzure.Storage.Table.TableOperation+<>c__DisplayClass65_0.<MergeImpl>b__0(RESTCommand<TableResult> cmd, Uri uri, UriQueryBuilder builder, HttpContent cnt, Nullable<int> serverTimeout, OperationContext ctx) in TableOperation.cs
      Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteAsyncInternal<T>(RESTCommand<T> cmd, IRetryPolicy policy, OperationContext operationContext, CancellationToken token) in Executor.cs
```

### How can we reproduce the problem in the simplest way?
I guess just try and update a row with the entity ETag set to either of the above values?

### Have you found a mitigation/solution?
No.

I have recently migrated a mothballed app from .NET 461 to .NET Core 2.2. This was all working previously.

The context is an ASP.NET Core 2.2 web application where I am using a custom identity/login store making use of Table Storage. With the local emulator and empty tables, the MS identity code appears to be creating a new user account in my Accounts table, then a new "login" (for my Google login) in my Identities table, then it seems to call update on the account entity it just created (for some reason) and that's when it fails.

When setting the `entity.ETag` to `null` I get:

```
Parameter name: Merge requires a valid ETag
```

**Comment by lukepuplett** [comment]
Not sure if this is related.

https://github.com/dotnet/corefx/issues/20994

**Comment by kfarmer-msft** [comment]
Hi @lukepuplett --

Sorry you're encountering issues there.  For table support, you're going to need to migrate to CosmosDB: https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Table.

**Comment by erezvani1529** [comment]
This is one of the breaking changes imposed by the stack and not the library; already listed in the breaking changes list:
https://github.com/Azure/azure-storage-net/blob/master/Blob/BreakingChanges.txt#L15

**Comment by lukepuplett** [comment]
Thanks for getting back to me.

Just for clarification, for local development, I should install the Cosmos DB emulator, then rip out my dependency on `WindowsAzure.Storage` and replace with the ref to `Microsoft.Azure.Cosmos.Table` and fix up code/connection strings etc.

Then for deployment, remove my original Table resource and replace with a new Cosmos DB resource configured with Table API?

**Comment by lukepuplett** [comment]
Okay so I did all this and ended up with the same `400 BadRequest`. The problem was the ETag being invalid or not matching. The Cosmos Table API should be returning `Precondition Failed` so it was misleading. Also the SDK ignores the ETag I set on insert and makes its own, returned in the `TableResult.Etag`.

I had to alter my data layer to return the inserted ETag and flow it back into my application layer and back down again for the second update. All done.
